tools/clang/include/clang/SPIRV/Structure.h

```diff
@@ -160,74 +229,8 @@ class SPIRVModule {
   inline void addFunction(std::unique_ptr<Function>);
 
 private:
-  /// \brief The struct representing a SPIR-V module header.
-  struct Header {
-    /// \brief Default constructs a SPIR-V module header with id bound 0.
-    Header();
-
-    /// \brief Feeds the consumer with all the SPIR-V words for this header.
-    void collect(const WordConsumer &consumer);
-
-    const uint32_t magicNumber;
-    const uint32_t version;
-    const uint32_t generator;
-    uint32_t bound;
-    const uint32_t reserved;
-  };
-
-  /// \brief The struct representing an extended instruction set.
-  struct ExtInstSet {
-    inline ExtInstSet(uint32_t id, std::string name);
-
-    const uint32_t resultId;
-    const std::string setName;
-  };
-
-  /// \brief The struct representing an entry point.
-  struct EntryPoint {
-    inline EntryPoint(spv::ExecutionModel, uint32_t id, std::string name,
-                      std::initializer_list<uint32_t> interface);
-
-    const spv::ExecutionModel executionModel;
-    const uint32_t targetId;
-    const std::string targetName;
-    const std::initializer_list<uint32_t> interfaces;
-  };
-
-  /// \brief The struct representing a debug name.
-  struct DebugName {
-    inline DebugName(uint32_t id, llvm::Optional<uint32_t> index,
-                     std::string targetName);
-
-    const uint32_t targetId;
-    const llvm::Optional<uint32_t> memberIndex;
-    const std::string name;
-  };
-
-  /// \brief The struct representing a deocoration and its target <result-id>.
-  struct DecorationIdPair {
-    inline DecorationIdPair(const Decoration &decor, uint32_t id);
-
-    const Decoration &decoration;
-    const uint32_t targetId;
-  };
-
-  /// \brief The struct representing a type and its <result-id>.
-  struct TypeIdPair {
-    inline TypeIdPair(const Type &ty, uint32_t id);
-
-    const Type &type;
-    const uint32_t resultId;
-  };
-
-  /// \brief The struct representing a constant and its type.
-  struct Constant {
-    inline Constant(const Type &ty, Instruction &&value);
-    const Type &type;
-    Instruction constant;
-  };
-
   Header header; ///< SPIR-V module header.
+  spv::ExecutionModel shaderKind; ///< Shader Type (Vertex, Fragment, etc.)
```

Comment:
One shader module can have multiple execution models. And execution modules are associated with entry point function ids. We should have vector of structs to represent the relationship. See https://www.khronos.org/registry/spir-v/specs/1.0/SPIRV.html#_validation_rules_for_shader_a_href_capability_capabilities_a